src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const datePartsArray = date.split(fromSeparator);
+
+  const dateParts = {};
+
+  fromFormat.forEach((part, index) => {
+    dateParts[part] = datePartsArray[index];
+  });
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  } else if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(dateParts['YY'], 10);
+
+    dateParts['YYYY'] =
+      year < 30 ? `20${dateParts['YY']}` : `19${dateParts['YY']}`;
+  }
+
+  const toSeparator = toFormat.pop();
```

Comment:
Similar to the issue with `fromFormat`, using `pop()` on `toFormat` modifies the array. Ensure that the separator is indeed the last element or consider a different method to handle separators.